code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -8,6 +8,30 @@
 	handle_casings = CYCLE_CASINGS
 	max_shells = 7
 	ammo_type = /obj/item/ammo_casing/a357
+	var/chamber_offset = 0 //how many empty chambers in the cylinder until you hit a round
+
+/obj/item/weapon/gun/projectile/revolver/verb/spin_cylinder()
+	set name = "Spin cylinder"
+	set desc = "Fun when you're bored out of your skull."
+	set category = "Object"
+
+	chamber_offset = 0
+	visible_message("<span class='warning'>\The [usr] spins the cylinder of \the [src]!</span>", \

```

Comment:
This is a bad call with the second argument. Call with `usr.` instead, otherwise, the `src` (the object to whom we send the second line) is the gun.